#goal
bool recorder_state
---
#result
bool recorder_result
---